test/cmd/apply.sh

```diff
@@ -438,6 +438,87 @@ run_kubectl_server_side_apply_tests() {
   # clean-up
   kubectl delete -f hack/testdata/pod.yaml "${kube_flags[@]:?}"
 
+    # Test apply migration
+
+  # Create a configmap in the cluster with client-side apply:
+  output_message=$(cat <<EOF | kubectl apply --server-side=false -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test
+data:
+  key: value
+  legacy: unused
+EOF
+  )
+
+  kube::test::if_has_string "${output_message}" 'configmap/test created'
+
+  # Apply the same manifest with --server-side flag, as per server-side-apply migration instructions:
+  output_message=$(cat <<EOF | kubectl apply --server-side -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test
+data:
+  key: value
+  legacy: unused
+EOF
+  )
+
+  kube::test::if_has_string "${output_message}" 'configmap/test serverside-applied'
+
+  # Apply the object a third time using server-side-apply, but this time removing
+  # a field. Old versions of kubectl would not allow the field to be removed"
+  output_message=$(cat <<EOF | kubectl apply --server-side -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test
+data:
+  key: value
+EOF
+  )
+
+  kube::test::if_has_string "${output_message}" 'configmap/test serverside-applied'
+
+  # Fetch the object and check to see that it does not have a field 'legacy'
+  kube::test::get_object_assert "configmap test" "{{ .data.legacy }}" '<no value>'
+
+  # CSA the object after it has been server-side-applied and had a field removed
+  # Add new key with client-side-apply.
+  output_message=$(cat <<EOF | kubectl apply --server-side=false -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test
+data:
+  key: value
```

Comment:
If you client-side apply without this key, does it get removed?

Reply:
I added a component to the integration test where client-side-apply removes a key added by server-side-apply.